tools/clang/test/HLSLFileCheck/d3dreflect/empty_broadcasting_nodes.hlsl

```diff
@@ -302,6 +302,6 @@ struct rec1
 [NodeDispatchGrid(2, 8, 10)]
 export void depth18part0_wg_63_nodes_seed_255(
   DispatchNodeInputRecord<rec0> InputyMcInputFace,
-  [MaxRecords(47)] [AllowSparseNodes] [NodeArraySize(2)] NodeOutput<rec1> OutputyMcOutputFace[2])
+  [MaxRecords(47)] [AllowSparseNodes] [NodeArraySize(2)] NodeOutput<rec1> OutputyMcOutputFace)
```

Comment:
Shouldn't this have changed `NodeOutput<rec1>` to `NodeOutputArray<rec1>`?  This highlights an undiagnosed problem: If you use `[NodeArraySize()]` attribute or the `[AllowSparseNodes]` on a non-array type.

Reply:
I've filed https://github.com/microsoft/DirectXShaderCompiler/issues/5810 to track the additional diagnostics.

Reply:
@tex3d, I've posted a change to update the test as you requested and filed an issue for the missing diagnostics.

Reply:
Great, but shouldn't we change this one to be well-formed, since the purpose isn't to test the upcoming diagnostics?

Reply:
Note that this `IOFlagsAndKind` is `NodeOutput` (6: ReadWrite|Output), instead of `NodeOutputArray` (22: ReadWrite|Output|NodeArray) as it should be:
```
// CHECK:      Outputs: <21:RecordArrayRef<IONode>[1]>  = {
// CHECK:        [0]: <1:IONode> = {
// CHECK:          IOFlagsAndKind: 6
```
And yet, some of the other properties included only apply to Array type (we should validate this in DxilValidation as well).